idunn/api/instant_answer.py

```diff
@@ -99,68 +126,9 @@ def get_instant_answer_single_place(place_id: str, lang: str):
     )
 
 
-async def get_instant_answer(
-    q: str = Query(..., title="Query string"), lang: str = Query("en", title="Language")
-):
-    """
-    Perform a query with result intended to be displayed as an instant answer
-    on *qwant.com*.
-
-    This should not be confused with "Get Places Bbox" as this endpoint will
-    run more restrictive checks on its results.
-    """
-    normalized_query = normalize(q)
-
-    if len(normalized_query) > ia_max_query_length:
-        raise NoInstantAnswerToDisplay
-
-    if normalized_query == "":
-        if settings["IA_SUCCESS_ON_GENERIC_QUERIES"]:
-            result = InstantAnswerResult(
-                places=[],
-                maps_url=maps_urls.get_default_url(),
-                maps_frame_url=maps_urls.get_default_url(no_ui=True),
-            )
-            return build_response(result, query=q, lang=lang)
-        raise NoInstantAnswerToDisplay
-
-    if lang in nlu_allowed_languages:
-        try:
-            intentions = await nlu_client.get_intentions(text=normalized_query, lang=lang)
-        except NluClientException:
-            intentions = []
-    else:
-        intentions = []
-
-    if not intentions:
-        bragi_response = await bragi_client.raw_autocomplete(
-            {"q": normalized_query, "lang": lang, "limit": 5}
-        )
-        geocodings = (feature["properties"]["geocoding"] for feature in bragi_response["features"])
-        geocodings = sorted(
-            (
-                (
-                    result_filter.rank_bragi_response(normalized_query, feature),
-                    feature,
-                )
-                for feature in geocodings
-                if result_filter.check_bragi_response(normalized_query, feature)
-            ),
-            key=lambda item: -item[0],  # sort by descending rank
-        )
-
-        if geocodings:
-            place_id = geocodings[0][1]["id"]
-            result = await run_in_threadpool(
-                get_instant_answer_single_place, place_id=place_id, lang=lang
-            )
-            return build_response(result, query=q, lang=lang)
-
-        raise NoInstantAnswerToDisplay
-
-    intention = intentions[0]
+async def get_instant_answer_intention(intention, query: str, lang: str):
     if not intention.filter.bbox:
-        raise NoInstantAnswerToDisplay
+        return no_instant_answer(query=query, lang=lang)
```

Comment:
I think I would have called the function `raise_no_instant_answer` instead, it would describe the intention slightly better. Here the `return` could lead to think that a value will actually be returned.

This is mostly subjective again so I wouldn't be surprised if you don't agree ^^